src/routes/Systems/Systems.js

```diff
@@ -1,51 +1,184 @@
+  /*tabletools state*/
+  const { items: itemsData } = useItemsData();
+  const tableState = useFullTableState();
+  const { current: { reload, resetSelection } = {} } = useStateCallbacks();
+  const { tableState: { selected } = {} } = tableState || {};
+
+  /*hooks*/
+  const exportConfig = useInventoryExport();
+
+  /*local state*/
+  const [removeHostsFromGroupModalOpen, setRemoveHostsFromGroupModalOpen] =
+    useState(false);
+  const [addHostGroupModalOpen, setAddHostGroupModalOpen] = useState(false);
+  const [currentSystem, setCurrentSystem] = useState({});
+  const [isRowAction, setIsRowAction] = useState(false);
+  const isKesselEnabled = useFeatureFlag('hbi.kessel-migration');
+
+  const selectedItems = useMemo(
+    () => itemsData?.filter(({ id }) => selected.includes(id)) || [],
+    [itemsData, selected],
+  );
+
+  const reloadData = useCallback(() => {
+    if (selectedItems.length > 0) {
+      resetSelection?.();
+      reload?.();
+    }
+  }, [reload, resetSelection, selectedItems]);
+
+  const toolbarActions = useToolbarActions(
+    selectedItems,
+    setAddHostGroupModalOpen,
+    setRemoveHostsFromGroupModalOpen,
+    setIsRowAction,
+  );
+
+  const {
+    itemsToDelete,
+    onConfirm,
+    dedicatedAction,
+    isModalOpen,
+    handleModalToggle,
+  } = useDeleteSystems(
+    itemsData,
+    reload,
+    resetSelection,
+    isRowAction ? [currentSystem.id] : selected,
+    setIsRowAction,
+  );
+
+  const { editModalOpen, onEditModalOpen, onSubmit } = useEditDisplayName(
+    currentSystem,
+    reload,
+    resetSelection,
+  );
+
+  const tableActions = useTableActions(
+    setCurrentSystem,
+    onEditModalOpen,
+    handleModalToggle,
+    setRemoveHostsFromGroupModalOpen,
+    setAddHostGroupModalOpen,
+    isKesselEnabled,
+    setIsRowAction,
+  );
+
+  return (
+    <>
+      <InventoryPageHeader />
+      <PageSection>
+        <SystemsTable
+          variant="compact"
+          columns={({
+            displayName,
+            workspace,
+            tags,
+            operatingSystem,
+            lastSeen,
+          }) => [displayName, workspace, tags, operatingSystem, lastSeen]}
+          filters={({
+            displayName,
+            status,
+            dataCollector,
+            rhcStatus,
+            tags,
+            systemType,
+            operatingSystem,
+          }) => ({
+            filterConfig: [
+              displayName,
+              status,
+              dataCollector,
+              rhcStatus,
+              tags,
+              systemType,
+              operatingSystem,
+            ],
+          })}
+          options={(defaultOptions) => ({
+            ...defaultOptions,
+            dedicatedAction,
+            actions: toolbarActions,
+            actionResolver: tableActions,
+            onSelect: true,
+          })}
+          toolbarProps={{
+            exportConfig,
+          }}
+        />
+
+        {addHostGroupModalOpen && (
```

Comment:
We could refactor this in the future and use `Outlet`[0] here and have the modals be sub-routes of `/systems`.

[0] https://reactrouter.com/6.30.1/components/outlet